### PR TITLE
fix(checkhealth): downgrade missing provider errors to warnings

### PR DESCRIPTION
**Pull request by dundargoc**
I'm trying to come up with a smooth way to get rid of the warnings altogether
(for myself) since I don't use providers and feel the flashing WARNING
distracting. I'm considering adding a global variable, something like
g:ignore_provider_error to make checkhealth usable again.

### CODE REVIEW

**Comment by clason** [comment]
Yeah, that has been irking me for a long time. While you're at it, maybe you could make all provider checks consistent in wording and formatting? For example, `cpan module not installed` is not really advice...
```
## Ruby provider (optional)
  - INFO: Ruby: ruby 2.6.8p205 (2021-07-07 revision 67951) [universal.arm64e-darwin21]
  - WARNING: `neovim-ruby-host` not found.
    - ADVICE:
      - Run `gem install neovim` to ensure the neovim RubyGem is installed.
      - Run `gem environment` to ensure the gem bin directory is in $PATH.
      - If you are using rvm/rbenv/chruby, try "rehashing".
      - See :help g:ruby_host_prog for non-standard gem installations.

## Node.js provider (optional)
  - INFO: Node.js: v17.7.1
  - WARNING: Missing "neovim" npm (or yarn) package.
    - ADVICE:
      - Run in shell: npm install -g neovim
      - Run in shell (if you use yarn): yarn global add neovim

## Perl provider (optional)
  - ERROR: perl provider error:
    - ADVICE:
      - "Neovim::Ext" cpan module is not installed
```

**Comment by dundargoc** [comment]
Heh, what do you know, there are already global variables for disabling each provider warning. Adding the following to your init.vim will silence all provider warnings:

```
let g:loaded_perl_provider=0
let g:loaded_ruby_provider=0
let g:loaded_python3_provider=0
let g:loaded_node_provider=0
```

I still intend to follow through @clason's great advice, just wanted to drop this as a reference in case someone else stumbles on this PR.

**Comment by clason** [comment]
These options don't just disable the warning, they disable the providers altogether. (Much startup time savings! Such wow!)

**Comment by dundargoc** [comment]
<details>
<summary>Click here to see the new checkhealth output.</summary>

## Python 3 provider (optional)
  - WARNING: No Python executable found that can `import neovim`. Using the first available executable for diagnostics.
  - WARNING: Could not load Python 3:
    /usr/bin/python3 does not have the "neovim" module.
    python3.10 not found in search path or not executable.
    /home/linuxbrew/.linuxbrew/bin/python3.9 does not have the "neovim" module.
    /usr/bin/python3.8 does not have the "neovim" module.
    python3.7 not found in search path or not executable.
    /usr/bin/python does not have the "neovim" module.
    - ADVICE:
      - See :help |provider-python| for more information.
      - You may disable this provider (and warning) by adding `let g:loaded_python3_provider = 0` to your init.vim
  - INFO: Executable: Not found

## Python virtualenv
  - OK: no $VIRTUAL_ENV

## Ruby provider (optional)
  - INFO: Ruby: ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux-gnu]
  - WARNING: `neovim-ruby-host` not found.
    - ADVICE:
      - Run `gem install neovim` to ensure the neovim RubyGem is installed.
      - Run `gem environment` to ensure the gem bin directory is in $PATH.
      - If you are using rvm/rbenv/chruby, try "rehashing".
      - See :help |g:ruby_host_prog| for non-standard gem installations.
      - You may disable this provider (and warning) by adding `let g:loaded_ruby_provider = 0` to your init.vim

## Node.js provider (optional)
  - INFO: Node.js: v14.19.0
  - WARNING: Missing "neovim" npm (or yarn) package.
    - ADVICE:
      - Run in shell: npm install -g neovim
      - Run in shell (if you use yarn): yarn global add neovim
      - You may disable this provider (and warning) by adding `let g:loaded_node_provider = 0` to your init.vim

## Perl provider (optional)
  - WARNING: "Neovim::Ext" cpan module is not installed
    - ADVICE:
      - See :help |provider-perl| for more information.
      - You may disable this provider (and warning) by adding `let g:loaded_node_provider = 0` to your init.vim
</details>

@clason Mind taking a look again?

**Comment by clason** [comment]
Yup, looks much better!

**Review by gpanders — APPROVED** [review]

File: runtime/autoload/health/provider.vim [hunk 1]
```diff
@@ -312,7 +312,10 @@ function! s:check_python() abort
 
   " No Python executable could `import neovim`, or host_prog_var was used.
   if !empty(pythonx_errors)
-    call health#report_error('Python provider error:', pythonx_errors)
+    call health#report_warn(pythonx_errors, ["See :help provider-python for more information.", 
+          \ "You may disable this provider (and warning) by adding `let g:loaded_python3_provider = 0` to your init.vim"
+          \])
+"If you wish to disable the Python provider (and this warning), then "
```
Comment (seandewar) [inline runtime/autoload/health/provider.vim:318]: this a stray string that's turned into a comment?

Comment (clason) [reply runtime/autoload/health/provider.vim:318]: can be removed as part of the typofix PR ;)

**Pull request merged**